java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -346,22 +360,18 @@ abstract class AbstractArcHost(
      */
     protected open suspend fun writeContextToStorage(arcId: String, context: ArcHostContext) {
         /** Serialize the [context] to storage in observed order asynchronously. */
-        if (!channel.isClosedForSend) {
-            var contextCopy: ArcHostContext? = ArcHostContext(
+        if (!contextSerializationChannel.isClosedForSend) {
```

Comment:
If we close the channel in ArcHost shutdown, wouldn't it be an error to arrive here with the channel closed? so we should just let it crash?

Reply:
crash is too bruteforced for a background serialization task.
I think a better way is just fallback.
```
val task = suspend { ... }

if (!contextSerializationChannel.isClosedForSend) channel.send(task)
else task()
```

Reply:
Refactored to support fallback!

Reply:
This is not a background task, this is the scheduling of a background task. 

The only way the channel will be closed is if we have closed it, which currently occurs in `shutdown`.

If `shutdown` has been called on the ArcHost, it means that the component owning the ArcHost has called `shutdown` on the Arc, but then explicitly called `startArc` or `stopArc` after that. This is incorrect usage, and should throw an exception.

Reply:
I actually see things differently: a startArc/stopArc is happening during ArcHost shutdown.
The shutdown has gone over the line of channel.close and racing startArc/stopArc just arrives at the checkpoint of isClosedForSend. I would like to choose to fallback instead of throwing an exception right away as the startArc/stopArc is happening during shutdown should be made not effective rather than throwing an exception.

As for the sequential case: a startArc/stopArc after a completed shutdown.
The assertion/exception/check should be put in the entry of startArc/stopArc rather than in the middle way of them.

Reply:
I think we should revisit this outside of this PR: Contractually, one should not call action methods on a component after closing it/shutting it down. Shutdown is a terminal operation. 

Calling `stopArc` or `startArc` after calling `shutdown` is a programmer error, so we should identify it and eliminate it.

Reply:
Yep, we can revisit it later.
I agree with shutdown IS a terminal op.

However a startArc can be started prior to a shutdown and they run at different threads racing in the ArcHost.
This is the situation I mentioned.

Currently on G3, only chronicle on->off will call the shutdown, so I think it's safe now.

Reply:
Ok I see your point.